src/gf.c

```diff
@@ -1084,20 +1090,20 @@ static int check_ambiguous_visitor(jl_typemap_entry_t *oldentry, struct typemap_
     jl_method_t *m = closure->newentry->func.method;
     jl_tupletype_t *sig = oldentry->sig;
     jl_value_t *isect = closure->match.ti;
-    if (sigs_eq(isect, (jl_value_t*)(closure->after ? sig : type), 1)) {
+    if (jl_types_equal(isect, (jl_value_t*)(closure->after ? sig : type))) {
         // we're ok if the new definition is actually the one we just
```

Comment:
Yes, we should eventually retire this condition (though I do miss printing a suggested signature to add). It could possibly be replaced by a rule in morespecific saying `New ⊓ Old == New` implies `New` is more specific.